validation and check in cron scaler for same start and end case

Ritikaa96 opened the pull request:
Signed-off-by: Ritikaa96 <ritika@india.nec.com>

_Provide a description of what has been changed_
In corn scaler, the target doesn't scale down when the start and end schedule are given the same input.

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Tests have been added
- [x] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)* https://github.com/kedacore/keda-docs/pull/515
- [x] Changelog has been updated

Fixes #2031

Ritikaa96 commented:
@zroubalik @ahmelsayed PTAL

ahmelsayed commented:
I'd add a note to the docs here (https://github.com/kedacore/keda-docs/blob/master/content/docs/2.5/scalers/cron.md) about this error case. @tomkerkhove what do you think?

tomkerkhove commented:
Agreed! Let's add it to all versions where CRON is available.

zroubalik reviewed (approved):
@Ritikaa96 looking good, could you please add a note to the Changelog (Improvements section) and add note to the docs as @ahmelsayed mentioned. Thanks!

Ritikaa96 commented:
> @Ritikaa96 looking good, could you please add a note to the Changelog (Improvements section) and add note to the docs as @ahmelsayed mentioned. Thanks!

Sure @zroubalik. Working on it.

Ritikaa96 commented:
@zroubalik @ahmelsayed @tomkerkhove 
Added the note in docs with https://github.com/kedacore/keda-docs/pull/515 
updated the changelog as well. PTAL

ahmelsayed commented:
Thanks @Ritikaa96
